src/output.rs

```diff
@@ -240,11 +240,18 @@ impl OutputData {
         OutputData(Arc::new(Mutex::new(OutputInfo::new(name))))
     }
 
+    /// Get the output scale factor.
     pub fn scale_factor(&self) -> i32 {
         let guard = self.0.lock().unwrap();
 
         guard.scale_factor
     }
+
+    /// Access the underlying [`OutputInfo`].
+    pub fn with_output_info<T, F: FnOnce(&OutputInfo) -> T>(&self, callback: F) -> T {
```

Comment:
We should probably warn about panics if you call this reentrantly.

Reply:
You mean within a callback again?

Reply:
Yes

Reply:
And we can't use `mut` for that, since data ruterns non-mut?

Reply:
I think it won't panic though, just a deadlock will occur?